doc/howtos/rdtraining/E_unittest.rst

```diff
@@ -0,0 +1,212 @@
+your own CI.
+
+Runbot
+------
+
+**Reference**: the documentation related to this topic can be found in
+`Runbot FAQ <https://runbot.odoo.com/doc>`__.
+
+Most of the tests are run on `Runbot <https://runbot.odoo.com>`__ every time a commit is pushed on
+GitHub.
+
+You can see the state of a commit/branch by filtering on the runbot dashboard.
+
+A **bundle** is created for each branch. A bundle consists of a configuration and contains the
+batches.
+
+A **batch** is a set of builds, depending on the parameters of the bundle.
+A batch is greened if all the builds are greened.
+
+A **build** is when we launch a server. It can be divided in sub-builds. Usually there are builds
+for the community version, the enterprise version (only if there is an enterprise branch but you
+can force the build), and the migration of the branch.
+A build is greened if every sub-build is greened.
+
+A **sub-build** only does some parts of what a full build does. It is used to speed up the CI
+process. Generally it is used to split the post install tests in 4 parallel instances.
+A sub-build is greened if all the tests are passing and there are no errors/warnings logged.
+
+.. note::
+  * Robodoo doesn't care what the changes were. It will always run the same tests, meaning it
+    will install all the modules. This means something might not work if runbot greened but your
+    changes depend on something you don't depend on.
+  * The localization modules are not installed on runbot (except the generic one), some modules
+    with external dependencies can be excluded also.
+  * There is a nightly build running additional tests, like module operations, localization, single
+    module installs, multi-builds for nondeterministic bugs, etc.
+    These are not kept in the standard CI to shorten the time of execution.
+
+You can also login on a build built by runbot. There are 3 users usable: `admin`, `demo` and
+`portal`. The password is the same as the login. This is useful to quickly test things on different
+versions without having to build it locally. The full logs are also available; these are used for
+monitoring.
+
+Robodoo
+-------
+
+You will most likely have to gain a little bit more experience before having the rights to summon
+robodoo, but here are a few remarks anyways.
+
+Robodoo is the guy spamming the CI status as tags on your PRs, but he is also the guy that kindly
+integrates your commits on the main repositories.
+
+When the last batch is greened, the reviewer can ask robodoo to merge your PR (actually it is more
+a `rebase` than a `merge`). It will then go to the mergebot.
+
+
+Mergebot
+--------
+
+`Mergebot <https://mergebot.odoo.com>`__ is the last testing phase before merging a PR.
+
+It will take the commits in your branch not yet present on the target, stage it and rerun the tests
+one more time, including the enterprise version even if you are only changing something in
+community.
+
```

Comment:
Maybe add some info about "Staging failed" error message?

More generally, there could be a part explaining what to do if the runbot is red. Check the logs, run the tests locally (maybe mention `--test-tags` and `--test-file`).